data/battle_scripts_1.s

```diff
@@ -6171,6 +6171,23 @@ BattleScript_MoveUsedPsychicTerrainPrevents::
 	waitmessage 0x40
 	goto BattleScript_MoveEnd
 
+BattleScript_GrassyTerrainLoop::
```

Comment:
Check for gBattleOutcome in the battlescript is redundant since hp can only go up.
HITMARKER_GRUDGE is redundant as well.

Also, making another command for checkgrassyterrainheal can be omitted as well, since you can check most of the things there directly in a script. 
jumpifstatus3
jumpifnotgrounded
manipulatedamage
etc.

Reply:
Removed the redundancies and replaced checkgrassyterrainheal with battlscripts 👍 . Only problem now is that Grassy Terrain heals fainted mons, which shouldn't happen.